test/chall-test-srv-client/client.go

```diff
@@ -353,14 +356,23 @@ func (c *Client) RemoveServfailResponse(host string) ([]byte, error) {
 	return resp, nil
 }
 
+var unencodedKeyAuthRegex = regexp.MustCompile(`^[A-Za-z0-9_-]+\.[A-Za-z0-9_-]+$`)
+
 // AddDNS01Response adds an ACME DNS-01 challenge response for the provided host
 // to the challenge test server's DNS interfaces. The provided value will be
 // served for TXT queries for _acme-challenge.<host>. Any failure returns an
 // error that includes both the relevant operation and the payload.
 func (c *Client) AddDNS01Response(host, value string) ([]byte, error) {
+	if !strings.HasPrefix(host, "_acme-challenge.") {
```

Comment:
For this helper I think it would be better to do both transforms unconditionally, and document them:

 - Prefix `_acme-challenge`.
 - Hash the value and store the RawURLEncoding of the hash.

I see the existing code, as ported from the Python, has a similar pattern where it adds the trailing `.` only if it's not present, but that's probably also unnecessary - I think every caller using ACME objects is going to omit the trailing `.`, so this client can add the trailing `.` unconditionally.

Reply:
that'd make chalsrv from doing things like [draft-ietf-acme-dns-account-label-00](https://www.ietf.org/archive/id/draft-ietf-acme-dns-account-label-00.html)
which needs different prefix